eshaz/icecast-metadata-js
# Metadata Recorder NodeJS

To run locally:
* install the latest version of NodeJS
* install the dependencies using `npm i`
* run `npm build-recorder` to create a packaged NodeJS application
* run `node recorder.js --help` for further options

To test:
* run `npm test` to test that parsing and cue file generation are working correctly.
* more tests to come.

FEATURES:
* NodeJS based metadata recorder that splits a constant bitrate Icecast stream into a data file and a metadata cue file.
* Metadata time accuracy up to 1/75 of a second.
* Configurable through the CLI or a JSON file.
* Ability to archive recordings on a periodic interval defined by a cron expression.

BREAKING CHANGES:
IcecastMetadataParser
* `onMetadataQueue` has been replaced with `onMetadata`

IcecastMetadataRecorder
* This has been considerably refactored with multiple parameter name changes and bug fixes. Please see the JSDocs for more details.
